pilot/pkg/model/sidecar.go

```diff
@@ -506,16 +505,12 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 					}
 					// If there is a port match, we should trim the service ports to the port specified by listener.
 					if portMatched {
+						ports := []*Port{}
 						for _, port := range s.Ports {
 							if port.Port == int(ilw.IstioListener.Port.GetNumber()) {
 								sc := s.DeepCopy()
-								sc.Ports = []*Port{
-									{
-										ilw.IstioListener.Port.GetName(),
-										int(ilw.IstioListener.Port.GetNumber()),
-										protocol.Instance(ilw.IstioListener.Port.GetProtocol()),
-									},
-								}
+								ports = append(ports, port)
+								sc.Ports = ports
```

Comment:
This includes all the ports?

Reply:
no.just matching ports

Reply:
earlier also it used to the same  - but copied from listener, now it just copies from service

Reply:
IC, could you move L508 down into the if branch?